code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -441,6 +441,25 @@
 	do_sparks(3, 1, src)
 	..()
 
+///Disassembling the bot in a civilized manner with a multitool
+/mob/living/simple_animal/bot/floorbot/disassemble()
+	on = FALSE
+	var/turf/Tsec = get_turf(src)
+	var/obj/item/storage/toolbox/mechanical/N = new /obj/item/storage/toolbox/mechanical(Tsec)
+	N.contents = list()
+	new /obj/item/assembly/prox_sensor(Tsec)
+	drop_part(robot_arm, Tsec)
+	while(amount)// Dumps the tiles into the appropriate sized stacks
+		if(amount >= 16)
+			var/obj/item/stack/tile/plasteel/T = new (Tsec)
+			T.amount = 16
+			amount -= 16
```

Comment:
I could've sworn these stacked till 50?

Reply:
The current `explode()` code of code/modules/mob/living/simple_animal/bot/floorbot.dm is :

`while(amount)// Dumps the tiles into the appropriate sized stacks
		if(amount >= 16)
			var/obj/item/stack/tile/plasteel/T = new (Tsec)
			T.amount = 16
			amount -= 16
		else
			var/obj/item/stack/tile/plasteel/T = new (Tsec)
			T.amount = amount
			amount = 0`
			
It may have been decided not to make it stack higher than 16 since the bot is getting destroyed ? I'll try to see if I can make them drop the exact amount of tiles that were left in the bot this evening.

Reply:
Sounds good!

Reply:
Made the bot drop the exact amount of floortiles it had left upon disassembly and made sure it does not create a stack of zero tiles.